### DISCUSSION TITLE
convex trigger shape cannot get SimulationEventCallback->onTrigger fired

### DISCUSSION THREAD

**Discussion by yananZZZZ** [NVIDIA-Omniverse/PhysX#144] [General]
Hi, as I trying to implement sensor behavior and I already make my convex shape work for collision by enable PxShapeFlag::eSIMULATION_SHAPE. but if I tried to make this shape as PxShapeFlag::eTRIGGER_SHAPE it will not fire onTrigger callback specified in my SimulationEventCallback. I did the same logic using box geometry, the onTrigger is fired when collision body enter and exit my sensor volume. I just wonder is there any special flag or other thing need to take care?
I dig more into the sdk found out this for convex case, the function "findTriggerContacts" overlap = overlapFunc() is false so there is no contact for trigger. Is this a bug or i did something wrong.

thank you.

**Comment by yananZZZZ** [comment]
![image](https://github.com/NVIDIA-Omniverse/PhysX/assets/122350362/1a189c50-e365-4967-a22a-f1189cf3bef5)
Hi
I would like to add more findings, inside "findTriggerContacts" I notice that overlapFunc varied for different type of geometry pair collide, in my case it would be convex VS box. But can see the picture above, the item for convex VS box is actually 0. Does it means this pair will do nothing? and it's different from GeomOverlapCallback_NotSupported, Why?
I changed my trigger and collision object both to convex which is defined in the table however the result is the same, which is not expected? any idea?

thank you.

**Comment by preist-nvidia** [comment]
The relevant function pointer is just above in the box type, FYI.

Are you raising both trigger and sim flags? Note that's illegal: https://nvidia-omniverse.github.io/PhysX/physx/5.1.3/_build/physx/latest/struct_px_shape_flag.html?highlight=etrigger_shape

Or do you mean you don't get the trigger event for a convex trigger shape?

**Reply by yananZZZZ** [reply]
Hi
I know that it's forbidden to raising trigger flag with simulation flag on. So, I already lower it before set trigger.  What's I mean is i cannot get my onTrigger fired after making the pair flag as "PxPairFlag::eTRIGGER_DEFAULT" and filter flag "PxFilterFlag::eDEFAULT" in my scene shader for a convex trigger shape.

thank you.

**Reply by preist-nvidia** [reply]
Ok, same question as the other issue, did you compare to our reference code in the snippets- this may help if you have not seen it yet: https://github.com/NVIDIA-Omniverse/PhysX/tree/release/104.2/physx/snippets/snippettriggers

**Reply by yananZZZZ** [reply]
Hi

    Thanks for your reply and I did a further investigation and found out that my trigger shape is rather small around 5mm box and my PxTolerancesScale length is set to be 2*e-6, however, i try to make scale even smaller, but it doesn't help.
   After I make my box bigger around 13mm, both trigger and scene query works. I suspect there is more things need to tune for tiny trigger shapes. It also me more confused that why trigger and collision have different precision as the same size collision works just fine.
    For trigger shape I can check all cook parameters but for scene query I have no idea.

thank you.
